r/src/array.cpp

```diff
@@ -403,11 +424,10 @@ SEXP DictionaryArray_to_Vector(arrow::DictionaryArray* dict_array) {
   auto indices = dict_array->indices();
 
   if (dict->type_id() != Type::STRING || indices->type_id() != Type::INT32) {
-    stop("Cannot convert Dictionary Array of type `%s`", dict_array->type()->ToString());
+    stop("Cannot convert Dictionary Array of type `%s` to R", dict_array->type()->ToString());
```

Comment:
Hm, probably want to bite the bullet on handling dictionary indices smaller than int32

Reply:
You can definitely use a smaller integer type when you are converting from R and the number of levels is small

Reply:
✅ handling int8, in16 and int32. No need for int64 since R indices are `int`